flash/core/model.py

```diff
@@ -276,7 +276,9 @@ class CheckDependenciesMeta(ABCMeta):
     def __new__(mcs, *args, **kwargs):
         result = ABCMeta.__new__(mcs, *args, **kwargs)
         if result.required_extras is not None:
-            result.__init__ = requires(result.required_extras)(result.__init__)
+            if isinstance(result.required_extras, str):
```

Comment:
@ethanwharris just curious why we have the wrapper and also use this new... can we simply use just one way?

Reply:
Yes, something that would be nice is to just expand the requires decorator so that if it gets a task it will also automatically wrap the `available_*` and `load_from_checkpoint` classmethods.

Reply:
ok, maybe another PR :)